### PR TITLE
[LUM-548] Upgrade dependencies and add Amino types

### PR DESCRIPTION
**Pull request by ThibaultJRD**
- Upgrade dependencies and fixes
- Add default Amino types function with all defaults types
- Add Dfract custom types to Amino
- Remove `Bech32` deprecated class and replace it by `fromBech32` and `toBech32` functions

### CODE REVIEW

File: src/client/LumClient.ts [hunk 1]
```diff
@@ -98,6 +99,8 @@ export class LumClient {
      */
     static connect = async (endpoint: string): Promise<LumClient> => {
         const tmClient = await Tendermint34Client.connect(endpoint);
+        console.log(createAminoTypes());
```
Comment (Segfaultd) [inline src/client/LumClient.ts:102]: This shouldn't be here

File: src/client/LumClient.ts [hunk 2]
```diff
@@ -274,7 +273,7 @@ export class LumClient {
      * @param doc document to sign
      */
     signTx = async (wallet: LumWallet | LumWallet[], doc: LumTypes.Doc): Promise<Uint8Array> => {
-        let wallets: LumWallet[] = [];
+        let wallets: LumWallet[];
```
Comment (Segfaultd) [inline src/client/LumClient.ts:276]: Any reason why this was deleted?

File: src/registry/aminoTypes.ts [hunk 3]
```diff
@@ -14,8 +24,22 @@ export function isAminoMsgDeposit(msg: AminoMsg): msg is AminoMsgDepositDfract {
...
+export const createDefaultAminoTypes = (): { [p: string]: AminoConverter | 'not_supported_by_chain' } => {
     return {
+        ...createBankAminoConverters(),
+        ...createAuthzAminoConverters(),
+        ...createDistributionAminoConverters(),
+        ...createGovAminoConverters(),
+        ...createStakingAminoConverters('lum'),
+        ...createIbcAminoConverters(),
+        ...createFeegrantAminoConverters(),
+        ...createVestingAminoConverters(),
```
Comment (Segfaultd) [inline src/registry/aminoTypes.ts:36]: Confident that all converters are here and not a single ones missing?

**Review by Segfaultd — CHANGES_REQUESTED** [review]

File: src/client/LumClient.ts (hunk 1 above)
Comment (ThibaultJRD) [reply src/client/LumClient.ts:102]: Oupsi

File: src/client/LumClient.ts (hunk 2 above)
Comment (ThibaultJRD) [reply src/client/LumClient.ts:274]: It's a redundant initializer, indeed, this variable is initialized in all cases just below

File: src/registry/aminoTypes.ts (hunk 3 above)
Comment (ThibaultJRD) [reply src/registry/aminoTypes.ts:36]: Yes, at least all we use for Wallet and DFract

File: src/client/LumClient.ts (hunk 2 above)
Comment (ThibaultJRD) [reply src/client/LumClient.ts]: Revamp all function

**Review by Segfaultd — APPROVED** [review]

**Pull request merged**